src/gluonts/model/seasonal_naive/_predictor.py

```diff
@@ -43,30 +43,35 @@ class SeasonalNaivePredictor(RepresentablePredictor):
 
     Parameters
     ----------
-    freq
-        Frequency of the input data
     prediction_length
-        Number of time points to predict
+        Number of time points to predict.
+    freq
+        Sampling frequency of the input data, used to infer ``season_length``
+        in case it is not provided.
     season_length
-        Length of the seasonality pattern of the input data
+        Seasonality used to make predictions. If not provided, it is inferred
+        from ``freq``.
     imputation_method
         The imputation method to use in case of missing values.
-        Defaults to `LastValueImputation` which replaces each missing
+        Defaults to :py:class:`LastValueImputation` which replaces each missing
         value with the last value that was not missing.
     """
 
     @validated()
     def __init__(
         self,
-        freq: str,
         prediction_length: int,
+        freq: Optional[str] = None,
         season_length: Optional[int] = None,
         imputation_method: Optional[
             MissingValueImputation
         ] = LastValueImputation(),
     ) -> None:
         super().__init__(prediction_length=prediction_length)
 
+        assert (freq is not None) or (
```

Comment:
Not sure this would be nicer.

```py
maybe.or_(season_length, freq).expect("You must provide one of `freq` or `season_length`")
```

Reply:
it's ignored; another option is that not both can be specified

Reply:
> Not sure this would be nicer.

not sure, I find the explicit assertion clearer

Reply:
I would keep the option of both parameters to be given, with `season_length` taking precedence.

Reply:
I think I actually like removing `freq` as a parameter and require setting `season_length` (why is it not called `seasonality`?).

That way things become much more explicit.

Reply:
> but with this modification this model would be the only one that explicitly requires the parameter season_length

I think that’s fine. The only thing common to all models really is that they have a fixed prediction length; in general, I think it’s fine that models each have their own set of options, related to how they work internally.

This model has a very simple logic, entirely determined by `season_length`: specifying this directly has no ambiguity. Adding indirection to translate `freq` only hides a little bit what the model will end up doing, with no added flexibility.

Then there are the choices for the default seasonality for [hourly](https://github.com/awslabs/gluonts/blob/3ccb6d377a5bf9b27de74a47cdab295f4d61f7a7/src/gluonts/time_feature/seasonality.py#L26) and [daily](https://github.com/awslabs/gluonts/blob/3ccb6d377a5bf9b27de74a47cdab295f4d61f7a7/src/gluonts/time_feature/seasonality.py#L27) data, which are completely arbitrary and end up tying together the behavior of different models or components for no reason.

In summary, I tend to agree that if you want to use the naive seasonal method, you should bring your own seasonality parameter. We do provide some heuristics to get that from frequency, and one is free to use them, but should do that explicitly.